src/map.cpp

```diff
@@ -829,89 +806,55 @@ bool Map::getPathMatching(const Creature& creature, std::vector<Direction>& dirL
 
 // AStarNodes
 
-AStarNodes::AStarNodes(uint32_t x, uint32_t y) : nodes(), openNodes()
+AStarNodes::AStarNodes(uint16_t x, uint16_t y) : nodes(), nodeMap()
 {
-	curNode = 1;
-	closedNodes = 0;
-	openNodes[0] = true;
-
-	AStarNode& startNode = nodes[0];
-	startNode.parent = nullptr;
-	startNode.x = x;
-	startNode.y = y;
-	startNode.f = 0;
-	nodeTable[(x << 16) | y] = nodes;
+	// Create our first node to check.
+	AStarNode* firstNode = new AStarNode;
+	firstNode->parent = nullptr;
+	firstNode->x = x;
+	firstNode->y = y;
+	firstNode->g = 0;
+	firstNode->f = 0;
+
+	// Add node to node vector and map
+	nodes.reserve(50);
+	nodes.push_back(firstNode);
```

Comment:
Upon further consideration about push_back being more suited for objects already created, there is the opportunity to construct in place here, probably more beneficial and more in line with modern C++ standards of RAII if we created the object with an initializer list, initializing the variables during construction... I think you would have to write your own constructor for the AStarNode class though, unless it happens to already have a contructor that uses an initializer list. Managing the data this way would allow the in-place construction benefits from emplace_back (including its performance increase) as well as the RAII.

This comment is **not** meant to be any kind of restrictive comment for review. I still stand by my approving review on this PR (the latest one).

Reply:
I don't see any issue using emplace_back. It does seem to be better for performance so I have added it. I also tested and it didn't seem to break anything.